soroban-sdk-auth/src/lib.rs

```diff
@@ -8,13 +8,48 @@ use crate::public_types::{
     SignaturePayloadV0,
 };
 
+/// Users of this module will need to pass a struct that implements NonceAuth
+/// NonceAuth should manage a nonce in the contracts storage
+///
+///
+/// Example implementation for NonceAuth -
+/// ```
+/// struct WrappedAuth(Signature);
+///
+/// impl NonceAuth for WrappedAuth {
+///
+/// fn read_nonce(e: &Env, id: Identifier) -> BigInt {
+///    let key = DataKey::Nonce(id);
+///    if let Some(nonce) = e.contract_data().get(key) {
+///       nonce.unwrap()
+///    } else {
+///        BigInt::zero(e)
+///    }
+/// }
+///
+/// fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt {
+///     let key = DataKey::Nonce(id.clone());
+///     let nonce = Self::read_nonce(e, id);
+///     e.contract_data()
+///         .set(key, nonce.clone() + BigInt::from_u32(e, 1));
+///     nonce
+/// }
+///
+/// fn get_keyed_auth(&self) -> &Signature {
+///     &self.0
+/// }
+/// }
+/// ```
 pub trait NonceAuth {
+    /// Return the nonce stored in the contract.
     fn read_nonce(e: &Env, id: Identifier) -> BigInt;
+    /// Return the nonce stored in the contract, and then increment it.
     fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt;
+    /// Return the Signature used for authorization.
     fn get_keyed_auth(&self) -> &Signature;
```

Comment:
We don't use the term `keyed` anymore in here, so I think this function name should be changed. Also, it returns a Signature, so maybe it could just be `signature()`?
```suggestion
    fn signature(&self) -> &Signature;
```

Reply:
Good catch. Updated to signature().